sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -59,14 +62,27 @@
  */
 public class Distinct<T> extends PTransform<PCollection<T>,
                                                     PCollection<T>> {
+  private boolean acrossPanes;
+
+  private Distinct(boolean acrossPanes) {
+    this.acrossPanes = acrossPanes;
+  }
+
   /**
    * Returns a {@code Distinct<T>} {@code PTransform}.
    *
    * @param <T> the type of the elements of the input and output
    * {@code PCollection}s
    */
   public static <T> Distinct<T> create() {
-    return new Distinct<T>();
+    return new Distinct<T>(false);
+  }
+
+  /**
+   * Ensures that elements in different panes are deduped.
+   */
+  public Distinct<T> acrossPanes() {
```

Comment:
I think we can enable the new behavior for non-default triggers or non-zero allowed lateness. In those cases, it can be assumed that current behavior is simply incorrect so there is no backwards-compatibility concern.